src/nnet3/nnet-simple-component.cc

```diff
@@ -143,7 +143,9 @@ void DropoutComponent::Backprop(const std::string &debug_info,
 
   KALDI_ASSERT(in_value.NumRows() == out_deriv.NumRows() &&
                in_value.NumCols() == out_deriv.NumCols());
-  in_deriv->AddMatMatDivMat(out_deriv, out_value, in_value);
+  CuMatrix<BaseFloat> in_deriv_inplace(in_deriv->NumRows(), in_deriv->NumCols());
```

Comment:
I'm not sure why you made these changes?  This shouldn't have been necessary.  Let me know if there was a test failure when you added the flag, the cause might be something else.